test/resources/qemu-guest-agent.py

```diff
@@ -0,0 +1,302 @@
+"""
+The qemu-guest-agent.py package replicates some keywords of the robot.library.Process. It uses virsh qemu-agent-command
+under the hood to interact with the guest-agent running on a VM. Due to behaviours of the qemu-guest-agent, return
+values are not necessarily congruent with the Process library.  This is because checking the status of a guest-agent's
+subprocess executes a destructive read on success. Keywords assume a RHEL-like guest OS, which is apparent in
+hardcoded command paths.
+
+The guest-agent package provides the following
+keywords:
+    - get_process_result
+    - start_process
+    - wait_for_process
+    - run_process
+
+If you are looking for keywords to control the guest VM itself, see ./libvirt.resource
+"""
+from __future__ import annotations  # Support for Python 3.7 and earlier
+
+import json
+from robot.libraries.BuiltIn import BuiltIn
+from robot.libraries.Process import Process, ExecutionResult
+
+
+def _do_kill(vm_name: str, pid: str | int, signal: int):
+    if not vm_name:
+        raise RuntimeError('vm name is not specified')
+    if not pid:
+        raise RuntimeError('pid is not specified')
+
+    agent_cmd_wrapper = {
+        'execute': 'guest-exec',
+        'arguments': {
+            'path': '/usr/bin/kill',
+            'arguments': [signal, pid],
+            'capture-output': True,
+        }
+    }
+    virsh_args = (f'--connect=qemu:///system qemu-agent-command --domain={vm_name} '
+                  f'--cmd={json.dumps(agent_cmd_wrapper)}')
+    result = Process().run_process('virsh', virsh_args)
+    if json.loads(result.stdout)['return']['exitcode'] != 0:
+        BuiltIn().log(f'failed to kill process {pid} on {vm_name}: {result.stderr}')
```

Comment:
It would be good to log all of the result output before using it on line 41. Then here if the exit code is not zero maybe raise an exception?

Reply:
fixed.  logs are now printed unconditionally.